docs/backends/akkahttp.rst

```diff
@@ -52,3 +52,14 @@ To receive the response body as a stream::
       .response(asStream[Source[ByteString, Any]])
       .send()
     
+
+Testing
+---------------
+
+For testing, you can create a backend using any `HttpRequest => Future[HttpResponse]` function - including a `Route`.
+
+That way, you can "mock" a server that the backend will talk to, without starting any actual server or making any HTTP calls.
+
+  val backend: SttpBackend[Future, Nothing] = {
+    AkkaHttpBackend.usingClient(system, http = AkkaHttpClient.fromAsyncHandler(Route.asyncHandler(Routes.route)))
```

Comment:
maybe we could have `AkkaHttpBackend.usingClient(system, AkkaHttpClient.fromRoute(Route))`? I think this will be the most common use-case.

Or even `AkkaHttpBackend.usingRoute(route)`?

I'm also thinking about names ... the route-based client is in fact a stub client -an alternative to `SttbBackendStub`. And that's when it will be used. Maybe the names should somehow reflect that? Just so that the reader of the code knows what this is about.

So maybe: `AkkaHttpClient.stubFromRoute` / `AkkaHttpClient.stubFromAsyncHandler`?

Reply:
I thought about that, but `usingRoute` will have like 6 implicit parameters (with defaults), same as `Route.asyncHandler`.

+1 for the stub naming